src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -162,6 +170,19 @@ public void renewClusterSlots(Jedis jedis) {
             }
           }
         }
+
+        if (startNodes != null) {
```

Comment:
If `getShuffledNodesPool` exists, it is almost impossible for the code to come here unless all master and slave nodes in the cluster are unserviceable.

Reply:
@yangbodong22011 OK, now I have clearer idea about #2643